File: social-map/src/main/java/com/mapsocial/service/MenuService.java
package com.mapsocial.service;

import com.mapsocial.dto.request.menu.CreateMenuRequest;
import com.mapsocial.dto.response.menu.MenuResponse;
import com.mapsocial.dto.request.menu.UpdateMenuRequest;

import java.util.UUID;

public interface MenuService {
    MenuResponse createMenu(CreateMenuRequest request, UUID sellerId);
    MenuResponse updateMenu(UUID menuId, UpdateMenuRequest request, UUID sellerId);
    void deleteMenu(UUID menuId, UUID sellerId);
    MenuResponse getMenuById(UUID menuId);
}

File: social-map-fe/src/services/menuService.js
import { api } from "./apiClient";

/**
 * Tạo menu mới
 * @param {Object} menuData
 * @returns {Promise<Object>} MenuResponse
 */
export const createMenu = async (menuData) => {
  return await api.post("/menus", menuData);
};

/**
 * Cập nhật menu
 * @param {string} menuId - UUID của menu
 * @param {Object} menuData
 * @returns {Promise<Object>} MenuResponse
 */
export const updateMenu = async (menuId, menuData) => {
  return await api.put(`/menus/${menuId}`, menuData);
};

/**
 * Xóa menu
 * @param {string} menuId - UUID của menu
 * @returns {Promise<void>}
 */
export const deleteMenu = async (menuId) => {
  return await api.delete(`/menus/${menuId}`);
};

/**
 * Lấy menu theo ID
 * @param {string} menuId - UUID của menu
 * @returns {Promise<Object>} MenuResponse
 */
export const getMenuById = async (menuId) => {
  return await api.get(`/menus/${menuId}`);
};
